src/util/sharing_node.h

```diff
@@ -43,7 +43,7 @@ Author: Daniel Poetzl
 #define SN_ASSERT_USE(v, b) SN_ASSERT(b)
 #else
 #define SN_ASSERT(b)
-#define SN_ASSERT_USE(v, b) v = v;
+#define SN_ASSERT_USE(v, b) static_cast<void>(v);
```

Comment:
Casting to `void` is not as portable as I thought (it's not going to shut up all compilers), in [here](https://herbsutter.com/2009/10/18/mailbag-shutting-up-compiler-warnings/#comment-1515) H. Sutter suggests something like

```
template<typename T>
void mark_as_used(const T&) {}
```

Although I suppose as long as it works for the compilers we're actually using I guess it's good enough.

Reply:
I think the full fix is to refactor the code so that this macro isn't necessary at all. There is a single use case at the moment, and a somewhat-dummy workaround would be adding a return value to the function.